tracer.go

```diff
@@ -122,9 +123,17 @@ func NewTracer(
 	}
 
 	if t.randomNumber == nil {
-		rng := utils.NewRand(time.Now().UnixNano())
+		seedGenerator := utils.NewRand(time.Now().UnixNano())
+		pool := sync.Pool{
+			New: func() interface{} {
+				return rand.NewSource(seedGenerator.Int63())
```

Comment:
Sorry to see this late, but I'm not sure I fully understand the logic here. Perhaps I'm missing something...

If the underlying issue is lock contention, how does moving to a `sync.Pool` solve the issue?  `Get()` still involves a mutex, correct?  And inside of that, we are still calling `Int63()` on a locked source, correct?

It *feels* like we want something that is lock-free. Have I missed something obvious?

Reply:
True there is a mutex on the pool, but the pool is returning a `rand.Source` which is not a locked resource. So there is no lock being held while the random numbers are being generated.

Reply:
Do we believe that getting a source is quicker than generating the numbers?

Reply:
From some basic tests, as described here https://github.com/jaegertracing/jaeger-client-go/pull/357#issuecomment-446683149, the benefit was more related to consistent results.

Not sure if @mandarjog has had a chance to try the latest version in the same test scenario to see whether it performed better?